src/Compilers/CSharp/Test/Semantic/SourceGeneration/GeneratorDriverTests_Attributes_FullyQualifiedName.cs

```diff
@@ -1580,7 +1580,7 @@ class XAttribute : System.Attribute
         Assert.Equal(IncrementalStepRunReason.Unchanged, runResult.TrackedSteps["collectedGlobalAliases_ForAttribute"].Single().Outputs.Single().Reason);
         Assert.Equal(IncrementalStepRunReason.Cached, runResult.TrackedSteps["compilationGlobalAliases_ForAttribute"].Single().Outputs.Single().Reason);
         Assert.Equal(IncrementalStepRunReason.Cached, runResult.TrackedSteps["allUpGlobalAliases_ForAttribute"].Single().Outputs.Single().Reason);
-        Assert.Equal(IncrementalStepRunReason.Modified, runResult.TrackedSteps["compilationUnit_ForAttribute"].Single().Outputs.Single().Reason);
+        Assert.Equal(IncrementalStepRunReason.New, runResult.TrackedSteps["compilationUnit_ForAttribute"].Single().Outputs.Single().Reason);
```

Comment:
> Do we have a concise example of the type of change that produces this behavior?

This test looks like a concise example. We have generator `FAWMN(XAttribute)`, start with `class C { }`, modify it to `[X] class C { }`, and the source generator produces new output for the class C due to modified input (adding attribute X). Previously, this would give `IncrementalStepRunReason.Modified`, now it gives `IncrementalStepRunReason.Modified`.